### PR TITLE
PC: Fix ssh key corner cases

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [poo#156643](https://progress.opensuse.org/issues/156643)
- Verification run: 
[Build5.32](https://pdostal-server.suse.cz/tests/overview?build=5.32&distri=sle&version=15-SP6) of sle-15-SP6-JeOS-for-OpenStack-Cloud.x86_64	[jeos-ltp-syscalls@64bit](https://pdostal-server.suse.cz/tests/5948)	
100 %
	5 minutes ago
[Build5.32](https://pdostal-server.suse.cz/tests/overview?build=5.32&distri=sle&version=15-SP6) of sle-15-SP6-JeOS-for-OpenStack-Cloud.x86_64	[jeos-containers-podman@64bit](https://pdostal-server.suse.cz/tests/5947)	
9 %
	5 minutes ago
[Build0536](https://pdostal-server.suse.cz/tests/overview?build=0536&distri=sle&version=15-SP6) of sle-15-SP6-EC2-BYOS.x86_64	[publiccloud_ltp_syscalls@ec2_t3.small](https://pdostal-server.suse.cz/tests/5946)	
100 %
	5 minutes ago
[Build0525](https://pdostal-server.suse.cz/tests/overview?build=0525&distri=sle&version=15-SP6) of sle-15-SP6-Azure-BYOS.x86_64	[publiccloud_ltp_syscalls@az_Standard_B2s](https://pdostal-server.suse.cz/tests/5945)	
100 %
	5 minutes ago
[Build5.32](https://pdostal-server.suse.cz/tests/overview?build=5.32&distri=sle&version=15-SP6) of sle-15-SP6-JeOS-for-OpenStack-Cloud.x86_64	[jeos-ltp-syscalls@64bit](https://pdostal-server.suse.cz/tests/5944)	
100 %
	5 minutes ago

### CODE REVIEW

**Comment by pdostal** [comment]
@mloviska please provide openstack run I can clone to verify this changes.

**Review by asmorodskyi — APPROVED** [review]
LGTM

File: tests/publiccloud/run_ltp.pm [hunk 1]
```diff
@@ -147,10 +147,15 @@ sub run {
     assert_script_run($log_start_cmd);
 
     # LTP command line preparation
+    assert_script_run('rpm -q python3-paramiko; true');
+    assert_script_run('zypper -n se -v paramiko; true');
     zypper_call("in -y python3-paramiko python3-scp");
```
Comment (ricardobranco777) [inline tests/publiccloud/run_ltp.pm:152]: Why?

Comment (pdostal) [reply tests/publiccloud/run_ltp.pm:152]: I don't know.

**Comment by m-dati** [comment]
Hi, please note that https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18801 also changed `get_ssh_private_key_path`: rebase needed

File: tests/publiccloud/run_ltp.pm (hunk 1 above)
Comment (pdostal) [reply tests/publiccloud/run_ltp.pm]: It now has a purpose. I commented it. Please re-check.

File: tests/publiccloud/run_ltp.pm [hunk 2]
```diff
@@ -147,10 +147,15 @@ sub run {
...
+    $instance->run_ssh_command(cmd => 'sudo sshd -T');
+    $instance->run_ssh_command(cmd => 'echo PubkeyAcceptedKeyTypes=+ssh-rsa | sudo tee -a /etc/ssh/sshd_config');
```
[the remark this answers is not in the document]
Comment (pdostal) [reply tests/publiccloud/run_ltp.pm]: Not applicable any more.

**Comment by pdostal** [comment]
> Hi, please note that #18801 also changed `get_ssh_private_key_path`: rebase needed

Rebased. Thank you.

**Review by ricardobranco777 — APPROVED** [review]

File: lib/publiccloud/provider.pm [hunk 3]
```diff
@@ -162,7 +162,7 @@ sub create_ssh_key {
     record_info($alg, "The $alg key will be generated.");
     if (script_run('test -f ' . $self->ssh_key) != 0) {
         assert_script_run('SSH_DIR=`dirname ' . $self->ssh_key . '`; mkdir -p $SSH_DIR');
-        assert_script_run('ssh-keygen -t ' . $alg . ' -q -N "" -C "" -m pem -f ' . $self->ssh_key);
+        assert_script_run('ssh-keygen -b 2048 -t ' . $alg . ' -q -N "" -C "" -m pem -f ' . $self->ssh_key);
```
Comment (m-dati) [inline lib/publiccloud/provider.pm:165]: Please why this 2048 value? 
From `man ssh-keygen`, for Ed25519 keys `-b` is ignored, and for RSA default is 3072 bits.

Comment (pdostal) [reply lib/publiccloud/provider.pm:165]: True, I will try to get rid of this one.

**Review by m-dati — APPROVED** [review]

**Pull request merged**